helix-view/src/editor.rs

```diff
@@ -103,14 +103,37 @@ pub struct Config {
     /// Whether to display infoboxes. Defaults to true.
     pub auto_info: bool,
     pub file_picker: FilePickerConfig,
+    /// Shape for cursor in each mode
+    pub cursor_shape: CursorShapeConfig,
+}
+
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+#[serde(transparent)]
+pub struct CursorShapeConfig(HashMap<Mode, CursorKind>);
+
+impl std::ops::Deref for CursorShapeConfig {
+    type Target = HashMap<Mode, CursorKind>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl Default for CursorShapeConfig {
+    fn default() -> Self {
+        Self(hashmap!(
+            Mode::Insert => CursorKind::Block,
+            Mode::Normal => CursorKind::Block,
+            Mode::Select => CursorKind::Block,
+        ))
+    }
 }
```

Comment:
I don't see merging code here, do we not need merging code in case use did `Mode::Insert` only?

Reply:
Default value for `CursorKind` is block, and we do an `unwrap_or_default()` when accessing the map (https://github.com/helix-editor/helix/pull/1154/files#diff-ca59dc0cbfb3ce30bc37926842acf5b65898146ac8c78c9e29229c933a480247R621-R627)

Reply:
I still think we should do merging in case we have different defaults. We don't need it now, but not sure if we would really want to default `CursorKind` to block without any given context.

Reply:
I think `CursorKind` `Default` impl as `Block` makes sense given that terminals default to block cursors.